Control/move_to_pose/move_to_pose_robot_class.py

```diff
@@ -0,0 +1,242 @@
+                 path_finder_controller):
+        """Inits Robot using its general specifications"""
+        self.name = name
+        self.color = color
+        self.MAX_LINEAR_SPEED = max_linear_speed
+        self.MAX_ANGULAR_SPEED = max_angular_speed
+        self.path_finder_controller = path_finder_controller
+        self.x_traj = []
+        self.y_traj = []
+        self.pose = Pose(0, 0, 0)
+        self.pose_start = Pose(0, 0, 0)
+        self.pose_target = Pose(0, 0, 0)
+        self.is_at_target = False
+
+    def set_start_target_poses(self, pose_start, pose_target):
+        self.pose_start = copy.copy(pose_start)
+        self.pose_target = pose_target
+        self.pose = pose_start
+
+    def move(self, dt):
+        self.x_traj.append(self.pose.x)
+        self.y_traj.append(self.pose.y)
+
+        x_diff = self.pose_target.x - self.pose.x
+        y_diff = self.pose_target.y - self.pose.y
+        rho = np.hypot(x_diff, y_diff)
+
+        if(rho < AT_TARGET_ACCEPTANCE_THRESHOULD):
+            self.is_at_target = True
+
+        # Restrict alpha and beta (angle differences) to the range
+        # [-pi, pi] to prevent unstable behavior e.g. difference going
+        # from 0 rad to 2*pi rad with slight turn
+
+        alpha = (np.arctan2(y_diff, x_diff)
+                 - self.pose.theta + np.pi) % (2 * np.pi) - np.pi
+        beta = (self.pose_target.theta - self.pose.theta - alpha
+                + np.pi) % (2 * np.pi) - np.pi
+
+        linear_velocity = self.path_finder_controller.Kp_rho * rho
+        angular_velocity = (self.path_finder_controller.Kp_alpha * alpha
+                            + self.path_finder_controller.Kp_beta * beta)
+
+        if alpha > np.pi / 2 or alpha < -np.pi / 2:
+            linear_velocity = -linear_velocity
+
+        if abs(linear_velocity) > self.MAX_LINEAR_SPEED:
+            linear_velocity = (np.sign(linear_velocity)
+                               * self.MAX_LINEAR_SPEED)
+
+        if abs(angular_velocity) > self.MAX_ANGULAR_SPEED:
+            angular_velocity = (np.sign(angular_velocity)
+                                * self.MAX_ANGULAR_SPEED)
+
+        self.pose.theta = self.pose.theta + angular_velocity * dt
+        self.pose.x = self.pose.x + linear_velocity * \
+            np.cos(self.pose.theta) * dt
+        self.pose.y = self.pose.y + linear_velocity * \
+            np.sin(self.pose.theta) * dt
+
+
+def run_simulation(robots):
+    """
+    rho is the distance between the robot and the target position
+    alpha is the angle to the target relative to the heading of the robot
+    beta is the angle between the robot's position and the target position plus
+    the target angle
+
+    Kp_rho*rho and Kp_alpha*alpha drive the robot along a line towards the
+    target
+    Kp_beta*beta rotates the line so that it is parallel to the target angle
+    """
+
+    global all_robots_are_at_target
+    global simulation_running
+
+    PLOT_WINDOW_SIZE_X = 20
```

Comment:
These const variables should go top of this file?

Reply:
- We need 'global all_robots_are_at_target' and 'global simulation_running' here. 
  - Note that these two global variables were already defined at line ~23 and ~24.
  - They are runtime flags. Aka. we need to _update_ them locally during simulation in order to terminate the simulation.
- I moved PLOT_WINDOW_SIZE_X = 20 to top of the file.